ext/pdo/php_pdo.h

```diff
@@ -55,11 +55,14 @@ PHP_MINFO_FUNCTION(pdo);
 
 #define LONG_CONST(c) (zend_long) c
 
-#define PDO_CONSTRUCT_CHECK \
+#define PDO_CONSTRUCT_CHECK_WITH_CLEANUP(cleanup) \
 	if (!dbh->driver) { \
 		zend_throw_error(NULL, "%s object is uninitialized", ZSTR_VAL(Z_OBJ(EX(This))->ce->name)); \
+		cleanup \
 		RETURN_THROWS(); \
 	} \
 
 
+#define PDO_CONSTRUCT_CHECK PDO_CONSTRUCT_CHECK_WITH_CLEANUP()
```

Comment:
This doesn't work: you're passing 0 arguments to a macro that takes 1 argument. I'd guess just passing a semicolon is enough.

Reply:
Clang and GCC transcripted my `()` as intended (with a blank cleanup).
But I'm embarrassed to stress-test compilers I've never had the pleasure to meet personally: how many of them will choke if someday someone wants a comma in its cleanup (as in `…_WITH_CLEANUP(int start = 0, end = 4; …)`: won't they complain I have too many parameters this time)? It's a too risky way.

So I reverted to a classical macro def in 3e7641459c0cbe5bb2ceb157901ca1abfd311f73. Let the IC see if MSVC, and the arginfo/zpp test, are happier with it.

Reply:
Sounds good, thanks